v0/destinations/signl4/utils.js

```diff
@@ -0,0 +1,57 @@
+const { constructPayload } = require("../../util");
+const { mappingConfig, ConfigCategory } = require("./config");
+
+/**
+ * This object is defined to map the fields being taken from UI with proper keys
+ */
+const propertyMappingObj = {
+  s4Service: "X-S4-Service",
+  s4Location: "X-S4-Location",
+  s4AlertingScenario: "X-S4-AlertingScenario",
+  s4ExternalID: "X-S4-ExternalID",
+  s4Status: "X-S4-Status"
+};
+
+/**
+ * This function is being used to populate payload. And it returns that prepared payload to be sent to the destination.
+ * @param {*} message
+ * @param {*} Config
+ * @returns
+ */
+const populatePayload = (message, Config) => {
+  const { properties, event } = message;
+  const { s4Filter, eventToTitleMapping } = Config;
+
+  let payload = constructPayload(
+    message,
+    mappingConfig[ConfigCategory.TRACK.name]
+  );
+
+  // populating all key value pairs of properties into the payload to be returned
+  payload = { ...payload, ...properties };
+
+  // Overriding event name with Customizable Title if avaiblable
+  eventToTitleMapping.forEach(mapping => {
```

Comment:
`eventToTitleMapping` is not a required field from Config, so if it's undefined then it gives an error here.

Reply:
Yes If it is not defined from the Config, then it gives an error.

Reply:
Done